flake.nix

```diff
@@ -0,0 +1,47 @@
+{
+  description = "REST API for any Postgres database";
+
+  inputs = {
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    flake-parts.inputs.nixpkgs-lib.follows = "nixpkgs";
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+  };
+
+  nixConfig = {
+    extra-substituters = "https://postgrest.cachix.org";
+    extra-trusted-public-keys = "postgrest.cachix.org:icgW4R15fz1+LqvhPjt4EnX/r19AaqxiVV+1olwlZtI=";
+  };
```

Comment:
Q: Does this mean that developers won't need to do an explicit `cachix use postgrest` before compiling?

Reply:
> Later, when we do our development environment via flake,

Once we do that, maybe we shouldn't solely rely on `nix flake check` for tests. I've found it really confusing because it depends on committing files to git and when no files are changed, the tests won't run again. I don't know if there are ways to avoid that.

Reply:
My comment is redundant given how nixConfig works. I'm glad you tagged me! 

You mention the user being asked: does the option require explicit configuration in non-interactive systems?

Reply:
> Turning tests into derivations has benefits - because tests are potentially cached, too. 

TIL that tests are turned into derivations with flakes, now it makes sense. Thanks for elaborating :smiley: 

> We can even add direnv support, then you don't need to run nix develop at all 

IMO direnv should be optional too. I tried it before but didn't have a good experience. I recall it failed sometimes and was slow loading. I'd be fine typing `nix develop/nix-shell` (although I avoid this by having preset projects with nvim-qt).

> and the whole "can I stick with my shell?" topic is done, too.

Cool. Great to know that flakes do bring some simplifications after all.